graphql_compiler/__init__.py

```diff
@@ -53,12 +58,48 @@ def graphql_to_match(schema, graphql_query, parameters, type_equivalence_hints=N
         query=insert_arguments_into_query(compilation_result, parameters))
 
 
+def graphql_to_sql(schema, graphql_query, parameters, compiler_metadata,
```

Comment:
Do you think in the long run (~v2.0) we'll want to include `type_equivalence_hints` into `compiler_metadata`? That way we could have a single compiler metadata object, and ideally make it the first argument so it can be partial-ed away easily. This is because that metadata and the schema don't really change, but the GraphQL query and parameters do -- so users could create a partial function `def compile_graphql(graphql_query: str, parameters: Dict[str, Any]) -> CompilationResult` for easier use.

In either case, I wouldn't change anything like that _now_, just thinking about the future.

Reply:
Yeah I think that's a great idea, the intent of my metadata object and the type equivalence hints are largely the same and unifying would make sense. That would seem a little less ergonomic to have metadata supplied first, but I like the intent. Maybe we would expect people to use `funcy.rpartial` since that's a project dependency?